pkg/broker/ingress/handler.go

```diff
@@ -120,29 +126,29 @@ func (h *handler) ServeHTTP(response nethttp.ResponseWriter, request *nethttp.Re
 		return
 	}
 	ns, broker := pieces[1], pieces[2]
-
 	event, msg, statusCode := h.toEvent(request)
 	if event == nil {
 		response.WriteHeader(statusCode)
 		response.Write([]byte(msg))
 		return
 	}
 
+	event.SetExtension(EventArrivalTime, cev2.Timestamp{Time: time.Now()})
+
 	ctx, cancel := context.WithTimeout(request.Context(), decoupleSinkTimeout)
 	defer cancel()
+	defer func() { h.reportMetrics(request.Context(), ns, broker, event, statusCode, time.Now()) }()
```

Comment:
Start time will be incorrect if it is calculated inside the defer. Instead make it an argument to the defer func.

Reply:
This is indeed a regression. Thanks!